File: dotnet/se3last-w3c/YosemiteTest.cs
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Edge;

namespace Se3LastW3C
{
    [TestClass]
    public class YosemiteTest : TestBase
    {
        [TestMethod]
        public void LateChrome()
        {
            var browserOptions = new ChromeOptions();
            browserOptions.PlatformName = "OS X 10.10";
            browserOptions.BrowserVersion = "87";
            var sauceOptions = new Dictionary<string, object>();
            browserOptions.AddAdditionalCapability("sauce:options", sauceOptions, true);

            StartDriver(browserOptions);
            ValidateGoogle();
        }

        [TestMethod]
        public void EarlyChrome()
        {
            var browserOptions = new ChromeOptions();
            browserOptions.PlatformName = "OS X 10.10";
            browserOptions.BrowserVersion = "75";
            var sauceOptions = new Dictionary<string, object>();
            browserOptions.AddAdditionalCapability("sauce:options", sauceOptions, true);

            StartDriver(browserOptions);
            ValidateGoogle();
        }

        [TestMethod]
        public void LateEdge()
        {
            var browserOptions = new EdgeOptions();
            browserOptions.PlatformName = "OS X 10.10";
            browserOptions.BrowserVersion = "81";
            var sauceOptions = new Dictionary<string, object>();
            browserOptions.AddAdditionalCapability("sauce:options", sauceOptions);

            StartDriver(browserOptions);
            ValidateGoogle();
        }

        [TestMethod]
        public void EarlyEdge()
        {
            var browserOptions = new EdgeOptions();
            browserOptions.PlatformName = "OS X 10.10";
            browserOptions.BrowserVersion = "79";
            var sauceOptions = new Dictionary<string, object>();
            browserOptions.AddAdditionalCapability("sauce:options", sauceOptions);

            StartDriver(browserOptions);
            ValidateGoogle();
        }
    }
}

File: java/se3last-w3c/src/test/java/com/saucelabs/platformconfigurator/se3lastw3c/YosemiteTest.java
package com.saucelabs.platformconfigurator.se3lastw3c;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.MutableCapabilities;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.edge.EdgeOptions;

public class YosemiteTest extends AbstractBaseTest {
    @DisplayName("Late Chrome")
    @Test
    public void lateChrome() {
        ChromeOptions browserOptions = new ChromeOptions();
        browserOptions.setCapability("platformName", "OS X 10.10");
        browserOptions.setCapability("browserVersion", "87");
        MutableCapabilities sauceOptions = new MutableCapabilities();
        browserOptions.setCapability("sauce:options", sauceOptions);

        startDriver(browserOptions);
        validateGoogle();
    }

    @DisplayName("Early Chrome")
    @Test
    public void earlyChrome() {
        ChromeOptions browserOptions = new ChromeOptions();
        browserOptions.setCapability("platformName", "OS X 10.10");
        browserOptions.setCapability("browserVersion", "75");
        MutableCapabilities sauceOptions = new MutableCapabilities();
        browserOptions.setCapability("sauce:options", sauceOptions);

        startDriver(browserOptions);
        validateGoogle();
    }

    @DisplayName("Late Edge")
    @Test
    public void lateEdge() {
        EdgeOptions browserOptions = new EdgeOptions();
        browserOptions.setCapability("platformName", "OS X 10.10");
        browserOptions.setCapability("browserVersion", "81");
        MutableCapabilities sauceOptions = new MutableCapabilities();
        browserOptions.setCapability("sauce:options", sauceOptions);

        startDriver(browserOptions);
        validateGoogle();
    }

    @DisplayName("Early Edge")
    @Test
    public void earlyEdge() {
        EdgeOptions browserOptions = new EdgeOptions();
        browserOptions.setCapability("platformName", "OS X 10.10");
        browserOptions.setCapability("browserVersion", "79");
        MutableCapabilities sauceOptions = new MutableCapabilities();
        browserOptions.setCapability("sauce:options", sauceOptions);

        startDriver(browserOptions);
        validateGoogle();
    }
}
